### PR TITLE
Bug 1216586 - Make sure we queue all mozbrowsermetachange events while waiting for the AppChrome to load. r=albertopq

### PR DESCRIPTION
**Pull request by etiennesegonzac**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7dd8886e135a2ea6d2929dc7a2300a8b0fa7d54a

File: apps/system/js/app_window.js [hunk 1]
```diff
@@ -1168,24 +1172,12 @@
...
-          var color = '';
-
-          // Otherwise, set it to the color that has been asked.
-          if (detail.type !== 'removed') {
-            color = detail.content;
-          }
-          this.themeColor = color;
-
-          this.publish('themecolorchange');
-          break;
```
Comment (etiennesegonzac) [inline apps/system/js/app_window.js]: removed because we do exactly the same (and override) in AppChrome

File: apps/system/js/app_window.js [hunk 2]
```diff
@@ -1202,7 +1194,6 @@
             }
           }
 
-          this.publish('themecolorchange');
```
Comment (etiennesegonzac) [inline apps/system/js/app_window.js]: removed because nobody is listening :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=547a2874ddd7a0566b5e96f591e74838b4f594a2

File: apps/system/js/app_window.js [hunk 3]
```diff
@@ -1168,24 +1172,15 @@
   AppWindow.prototype._handle_mozbrowsermetachange =
     function aw__handle_mozbrowsermetachange(evt) {
 
-      var detail = this.metachangeDetail = evt.detail;
+      var detail = evt.detail;
+      if (!this.appChrome) {
+        if (!this.metachangeDetails) {
```
Comment (albertopq) [inline apps/system/js/app_window.js]: this.metachangeDetails = this.metachangeDetails || []; ?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6490380bb01522bdb60c3ee13f565f4c55786806

**Pull request merged**